src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -389,9 +386,24 @@ public void Abort()
                         using (cancellationToken.Register(s => ((Socket)s).Dispose(), socket))
                         using (_abortSource.Token.Register(s => ((Socket)s).Dispose(), socket))
                         {
-                            await socket.ConnectAsync(address, port).ConfigureAwait(false);
+                            try
+                            {
+                                await socket.ConnectAsync(address, port).ConfigureAwait(false);
+                            }
+                            catch (ObjectDisposedException)
+                            {
+                                // If the socket was disposed because cancellation was requested, translate the exception
+                                // into a new OperationCanceledException.  Otherwise, let the original ObjectDisposedexception propagate.
+                                CancellationToken token = cancellationToken.IsCancellationRequested ? cancellationToken : _abortSource.Token;
+                                if (token.IsCancellationRequested)
+                                {
+                                    throw new OperationCanceledException(token);
```

Comment:
Yeah, intentional, since the ODE doesn't mean much here. I guess it wouldn't hurt to carry it as the inner inner exception, though.